File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/lineage_relationship.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class LineageRelationship(object):
    """
    Declares how two elements of object lineage are related.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new LineageRelationship object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param from_node_id:
            The value to assign to the from_node_id property of this LineageRelationship.
        :type from_node_id: str

        :param to_node_id:
            The value to assign to the to_node_id property of this LineageRelationship.
        :type to_node_id: str

        :param type:
            The value to assign to the type property of this LineageRelationship.
        :type type: str

        :param provider_type:
            The value to assign to the provider_type property of this LineageRelationship.
        :type provider_type: str

        :param properties:
            The value to assign to the properties property of this LineageRelationship.
        :type properties: dict(str, dict(str, str))

        """
        self.swagger_types = {
            'from_node_id': 'str',
            'to_node_id': 'str',
            'type': 'str',
            'provider_type': 'str',
            'properties': 'dict(str, dict(str, str))'
        }

        self.attribute_map = {
            'from_node_id': 'fromNodeId',
            'to_node_id': 'toNodeId',
            'type': 'type',
            'provider_type': 'providerType',
            'properties': 'properties'
        }

        self._from_node_id = None
        self._to_node_id = None
        self._type = None
        self._provider_type = None
        self._properties = None

    @property
    def from_node_id(self):
        """
        Gets the from_node_id of this LineageRelationship.
        Object key of source lineage element.


        :return: The from_node_id of this LineageRelationship.
        :rtype: str
        """
        return self._from_node_id

    @from_node_id.setter
    def from_node_id(self, from_node_id):
        """
        Sets the from_node_id of this LineageRelationship.
        Object key of source lineage element.


        :param from_node_id: The from_node_id of this LineageRelationship.
        :type: str
        """
        self._from_node_id = from_node_id

    @property
    def to_node_id(self):
        """
        Gets the to_node_id of this LineageRelationship.
        Object key of target lineage element.


        :return: The to_node_id of this LineageRelationship.
        :rtype: str
        """
        return self._to_node_id

    @to_node_id.setter
    def to_node_id(self, to_node_id):
        """
        Sets the to_node_id of this LineageRelationship.
        Object key of target lineage element.


        :param to_node_id: The to_node_id of this LineageRelationship.
        :type: str
        """
        self._to_node_id = to_node_id

    @property
    def type(self):
        """
        Gets the type of this LineageRelationship.
        Type of the relationship.


        :return: The type of this LineageRelationship.
        :rtype: str
        """
        return self._type

    @type.setter
    def type(self, type):
        """
        Sets the type of this LineageRelationship.
        Type of the relationship.


        :param type: The type of this LineageRelationship.
        :type: str
        """
        self._type = type

    @property
    def provider_type(self):
        """
        Gets the provider_type of this LineageRelationship.
        Type of the relationship.


        :return: The provider_type of this LineageRelationship.
        :rtype: str
        """
        return self._provider_type

    @provider_type.setter
    def provider_type(self, provider_type):
        """
        Sets the provider_type of this LineageRelationship.
        Type of the relationship.


        :param provider_type: The provider_type of this LineageRelationship.
        :type: str
        """
        self._provider_type = provider_type

    @property
    def properties(self):
        """
        Gets the properties of this LineageRelationship.
        A map of maps that contains the properties which are specific to the entity type. Each entity type
        definition defines it's set of required and optional properties. The map keys are category names and the
        values are maps of property name to property value. Every property is contained inside of a category. Most
        data entities have required properties within the \"default\" category.
        Example: `{\"properties\": { \"default\": { \"key1\": \"value1\"}}}`


        :return: The properties of this LineageRelationship.
        :rtype: dict(str, dict(str, str))
        """
        return self._properties

    @properties.setter
    def properties(self, properties):
        """
        Sets the properties of this LineageRelationship.
        A map of maps that contains the properties which are specific to the entity type. Each entity type
        definition defines it's set of required and optional properties. The map keys are category names and the
        values are maps of property name to property value. Every property is contained inside of a category. Most
        data entities have required properties within the \"default\" category.
        Example: `{\"properties\": { \"default\": { \"key1\": \"value1\"}}}`


        :param properties: The properties of this LineageRelationship.
        :type: dict(str, dict(str, str))
        """
        self._properties = properties

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/LineageRelationship.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Declares how two elements of object lineage are related.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=LineageRelationship.Builder.class)

public final class LineageRelationship  {
    @Deprecated
    @java.beans.ConstructorProperties({"fromNodeId", "toNodeId", "type", "providerType", "properties"})
    public LineageRelationship(String fromNodeId, String toNodeId, String type, String providerType, java.util.Map<String, java.util.Map<String, String>> properties) {
        super();
        this.fromNodeId = fromNodeId;
        this.toNodeId = toNodeId;
        this.type = type;
        this.providerType = providerType;
        this.properties = properties;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Object key of source lineage element.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("fromNodeId")
private String fromNodeId;

        /**
         * Object key of source lineage element.
         * @param fromNodeId the value to set
         * @return this builder
         **/
        

public Builder fromNodeId(String fromNodeId) {
    this.fromNodeId = fromNodeId;
    return this;
}
            /**
     * Object key of target lineage element.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("toNodeId")
private String toNodeId;

        /**
         * Object key of target lineage element.
         * @param toNodeId the value to set
         * @return this builder
         **/
        

public Builder toNodeId(String toNodeId) {
    this.toNodeId = toNodeId;
    return this;
}
            /**
     * Type of the relationship.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("type")
private String type;

        /**
         * Type of the relationship.
         * @param type the value to set
         * @return this builder
         **/
        

public Builder type(String type) {
    this.type = type;
    return this;
}
            /**
     * Type of the relationship.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("providerType")
private String providerType;

        /**
         * Type of the relationship.
         * @param providerType the value to set
         * @return this builder
         **/
        

public Builder providerType(String providerType) {
    this.providerType = providerType;
    return this;
}
            /**
     * A map of maps that contains the properties which are specific to the entity type. Each entity type
* definition defines it's set of required and optional properties. The map keys are category names and the
* values are maps of property name to property value. Every property is contained inside of a category. Most
* data entities have required properties within the "default" category.
* Example: {@code {"properties": { "default": { "key1": "value1"}}}}
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("properties")
private java.util.Map<String, java.util.Map<String, String>> properties;

        /**
         * A map of maps that contains the properties which are specific to the entity type. Each entity type
* definition defines it's set of required and optional properties. The map keys are category names and the
* values are maps of property name to property value. Every property is contained inside of a category. Most
* data entities have required properties within the "default" category.
* Example: {@code {"properties": { "default": { "key1": "value1"}}}}
* 
         * @param properties the value to set
         * @return this builder
         **/
        

public Builder properties(java.util.Map<String, java.util.Map<String, String>> properties) {
    this.properties = properties;
    return this;
}


        public LineageRelationship build() {
            LineageRelationship model = new LineageRelationship(this.fromNodeId
                , this.toNodeId
                , this.type
                , this.providerType
                , this.properties);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(LineageRelationship model) {
                this.fromNodeId(model.getFromNodeId());
    this.toNodeId(model.getToNodeId());
    this.type(model.getType());
    this.providerType(model.getProviderType());
    this.properties(model.getProperties());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Object key of source lineage element.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("fromNodeId")
    private final String fromNodeId;

        /**
     * Object key of source lineage element.
     * @return the value
     **/
    
    public String getFromNodeId() {
        return fromNodeId;
    }


        /**
     * Object key of target lineage element.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("toNodeId")
    private final String toNodeId;

        /**
     * Object key of target lineage element.
     * @return the value
     **/
    
    public String getToNodeId() {
        return toNodeId;
    }


        /**
     * Type of the relationship.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("type")
    private final String type;

        /**
     * Type of the relationship.
     * @return the value
     **/
    
    public String getType() {
        return type;
    }


        /**
     * Type of the relationship.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("providerType")
    private final String providerType;

        /**
     * Type of the relationship.
     * @return the value
     **/
    
    public String getProviderType() {
        return providerType;
    }


        /**
     * A map of maps that contains the properties which are specific to the entity type. Each entity type
* definition defines it's set of required and optional properties. The map keys are category names and the
* values are maps of property name to property value. Every property is contained inside of a category. Most
* data entities have required properties within the "default" category.
* Example: {@code {"properties": { "default": { "key1": "value1"}}}}
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("properties")
    private final java.util.Map<String, java.util.Map<String, String>> properties;

        /**
     * A map of maps that contains the properties which are specific to the entity type. Each entity type
* definition defines it's set of required and optional properties. The map keys are category names and the
* values are maps of property name to property value. Every property is contained inside of a category. Most
* data entities have required properties within the "default" category.
* Example: {@code {"properties": { "default": { "key1": "value1"}}}}
* 
     * @return the value
     **/
    
    public java.util.Map<String, java.util.Map<String, String>> getProperties() {
        return properties;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("LineageRelationship(");
        sb.append("fromNodeId=").append(String.valueOf(this.fromNodeId));
        sb.append(", toNodeId=").append(String.valueOf(this.toNodeId));
        sb.append(", type=").append(String.valueOf(this.type));
        sb.append(", providerType=").append(String.valueOf(this.providerType));
        sb.append(", properties=").append(String.valueOf(this.properties));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof LineageRelationship)) {
            return false;
        }

        LineageRelationship other = (LineageRelationship) o;
        return java.util.Objects.equals(this.fromNodeId, other.fromNodeId) &&
            java.util.Objects.equals(this.toNodeId, other.toNodeId) &&
            java.util.Objects.equals(this.type, other.type) &&
            java.util.Objects.equals(this.providerType, other.providerType) &&
            java.util.Objects.equals(this.properties, other.properties);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.fromNodeId == null ? 43 : this.fromNodeId.hashCode());
        result = (result * PRIME) + (this.toNodeId == null ? 43 : this.toNodeId.hashCode());
        result = (result * PRIME) + (this.type == null ? 43 : this.type.hashCode());
        result = (result * PRIME) + (this.providerType == null ? 43 : this.providerType.hashCode());
        result = (result * PRIME) + (this.properties == null ? 43 : this.properties.hashCode());
        return result;
    }


}
